en/getting-started/first-game/index.md

```diff
@@ -1,153 +1,470 @@
+    }
+
+
+    update(dt: number): void {
+        if( this._startJump){
+
+        }
+    }
+}
+```
+
+Fascinating, now we have all the data required to calculate the character's manipulation, then we can move it in the `update` method.
+
+Remember the Newton problem we talked about above? It's time to really solve this problem.
+
+We need to first check if the entire jump duration (_curJumpTime) has exceeded our predefined time (_jumpTime).
+
+The test here is simple: we have set _curJumpTime to 0 at input time, so `update` only needs to:
+
+```
+Jump time = last jump time + frame interval
+```
+
+The `update` method happens to provide a parameter like `deltaTime` as the frame interval.
+
+So we just add `_curJumpTime` to `deltaTime` and that's it:
+
+```ts
+update (deltaTime: number) {
+    if (this._startJump) {
+        this._curJumpTime += deltaTime;        
+    }
+}
+```
+
+> You can see that we ues the '+=' operator, which is equivalent to `this._curJumpTime = this._curJumpTime + deltaTime`.
+
+When `_curJumpTime` is greater than our pre-defined `_jumpTime`, this means that the jump is over, so we need two branches to handle the end-of-jump and in-jump states, respectively:
+
+```ts
+update (deltaTime: number) {
+    if (this._startJump) {
+        this._curJumpTime += deltaTime; 
+        if (this._curJumpTime > this._jumpTime) { // Jump ends
+            
+            
+        } else { // Jumping
+                        
+        }
+    }
+}
+```
+
+Finally, we refine their logic for the different branches: t
+
+- End of jump: ends the jumping process, forcing the character to be displaced to the target position
+- During the jump: moves the character forward according to the velocity
+
+So our `update` method will look like this:
+
+```ts
+update (deltaTime: number) {
+    if (this._startJump) {
+        this._curJumpTime += deltaTime.
+        if (this._curJumpTime > this._jumpTime) { // end of jump
+            // end
+            this.node.setPosition(this._targetPos); // force move to target position
+            this._startJump = false; // mark the end of the jump
+        } else { // jump in progress
+            // tween
+            this.node.getPosition(this._curPos); // Get the current position 
+            this._deltaPos.x = this._curJumpSpeed * deltaTime; // calculate the length of this frame that should be displaced
+            Vec3.add(this._curPos, this._curPos, this._deltaPos); // add the current position to the length of the displacement
+            this.node.setPosition(this._curPos); // set the position after displacement
+        }
+    }
+}
+```
+
+Great! You have completed the core of this game - character control.
+
+If you still find this explanation a bit difficult, please let us know by clicking on [Get help and support](../support.md) and let us know.
+
+#### Complete PlayerController
+
+We already have the complete PlayerController code, so we just need to hook it up to a node to make it work.
```

Comment:
hook it up to -> attach it to

"hook it up to"  implies the process is complex.  
however, "attach it to" is a more general term and can refer to situations from simple to complex.